test/jdk/java/lang/reflect/code/TestStringConcatTransform.java

```diff
@@ -0,0 +1,204 @@
+        valMap = new HashMap<>();
+        valMap.put(byte.class, (byte) 42);
+        valMap.put(short.class, (short) 42);
+        valMap.put(int.class, 42);
+        valMap.put(long.class, (long) 42);
+        valMap.put(float.class, 42f);
+        valMap.put(double.class, 42d);
+        valMap.put(char.class, 'z');
+        valMap.put(boolean.class, false);
+
+        valMap.put(Byte.class, (byte) 42);
+        valMap.put(Short.class, (short) 42);
+        valMap.put(Integer.class, 42);
+        valMap.put(Long.class, (long) 42);
+        valMap.put(Float.class, 42f);
+        valMap.put(Double.class, 42d);
+        valMap.put(Character.class, 'z');
+        valMap.put(Boolean.class, false);
+
+        valMap.put(Object.class, new Object() {
+            @Override
+            public String toString() {
+                return "I'm a test string.";
+            }
+        });
+        valMap.put(TestObject.class, new TestObject());
+        valMap.put(String.class, TESTSTR);
+    }
+
+    public static final class TestObject {
+        TestObject() {
+        }
+
+        @Override
+        public String toString() {
+            return "TestObject String";
+        }
+    }
+
+    @Test(dataProvider = "getClassMethods")
+    public void testModelTransform(@NoInjection Method method) {
+        CoreOp.FuncOp model = method.getCodeModel().orElseThrow();
+        CoreOp.FuncOp f_transformed = model.transform(new LinearStringContextTransformer());
+        Object[] args = prepArgs(method);
+
+        f_transformed.writeTo(System.out);
+
+        var interpreted = Interpreter.invoke(model, args);
+        var transformed_interpreted = Interpreter.invoke(f_transformed, args);
+
+        Assert.assertEquals(interpreted, transformed_interpreted);
+
+    }
+
+    @Test(dataProvider = "getClassMethods")
+    public void testSSAModelTransform(@NoInjection Method method) {
+        CoreOp.FuncOp model = method.getCodeModel().orElseThrow();
+        CoreOp.FuncOp transformed_model = model.transform(new LinearStringContextTransformer());
+        CoreOp.FuncOp ssa_model = generateSSA(model);
+        CoreOp.FuncOp ssa_transformed_model = ssa_model.transform(new LinearStringContextTransformer());
+        Object[] args = prepArgs(method);
+
+        var model_interpreted = Interpreter.invoke(model, args);
+        var transformed_model_interpreted = Interpreter.invoke(transformed_model, args);
+        var ssa_interpreted = Interpreter.invoke(ssa_model, args);
+        var ssa_transformed_interpreted = Interpreter.invoke(ssa_transformed_model, args);
+        try {
+            var jvm_interpreted = method.invoke(null, args);
+            Assert.assertEquals(model_interpreted, transformed_model_interpreted);
+            Assert.assertEquals(transformed_model_interpreted, ssa_interpreted);
+            Assert.assertEquals(ssa_interpreted, ssa_transformed_interpreted);
+            Assert.assertEquals(ssa_transformed_interpreted, jvm_interpreted);
```

Comment:
Won't the asserts errors get wrapped in RuntimeException? suggest you pull 'em out of. e.g.,
```
  String jvm_interpreted;
  try {
    jvm_interpreted = method.invoke(null, args);
  } catch (...) { ... }

  Assert....
```

Reply:
You're right, they should probably come out just so it's clear they're not getting caught. I believe TestNG subclasses from `Error`.

Reply:
...Ah misunderstood, yes they're getting wrapped and rethrown in the RuntimeException.